### PR TITLE
Use table info

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
Fix https://github.com/pingcap/br/pull/230#issuecomment-610798266

### What is changed and how it works?
1. use `ddljob.TableInfo` and `ddljob.DBInfo` to create tables and schema in incremental restore.
2. set `config.AlterPrimaryKey` to true so we can execute drop/add pk DDL in incremental restore.

### Check List 

Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: pkg/restore/db.go [hunk 1]
```diff
@@ -45,7 +45,27 @@ func NewDB(g glue.Glue, store kv.Storage) (*DB, error) {
...
+	dbInfo := ddlJob.BinlogInfo.DBInfo
+	switch ddlJob.Type {
+	case model.ActionCreateSchema:
+		err = db.se.CreateDatabase(ctx, dbInfo)
+		if err != nil {
+			log.Error("create database failed", zap.Stringer("db", dbInfo.Name), zap.Error(err))
+		}
+		return errors.Trace(err)
+	case model.ActionCreateTable:
+		err = db.se.CreateTable(ctx, model.NewCIStr(ddlJob.SchemaName), tableInfo)
```
Comment (kennytm) [inline pkg/restore/db.go:58]: we should expose the "IF NOT EXISTS" parameter

File: tests/br_incompatible_tidb_config/run.sh [hunk 2]
```diff
@@ -29,17 +29,30 @@ run_sql "create schema $DB;"
...
+INCREMENTAL_TABLE="t1inc"
 
 run_sql "create table $DB.$TABLE (a int primary key, b int unique);"
 run_sql "insert into $DB.$TABLE values (42, 42);"
 
 # backup
 run_br --pd $PD_ADDR backup db --db "$DB" -s "local://$TEST_DIR/$DB$TABLE"
 
+run_sql "create table $DB.$INCREMENTAL_TABLE (a int primary key, b int unique);"
+run_sql "insert into $DB.$INCREMENTAL_TABLE values (42, 42);"
```
Comment (kennytm) [inline tests/br_incompatible_tidb_config/run.sh:41]: please include a test case for [`CREATE TABLE … LIKE`](https://dev.mysql.com/doc/refman/8.0/en/create-table-like.html).

File: pkg/restore/db.go (hunk 1 above)
Comment (3pointer) [reply pkg/restore/db.go:58]: I didn't get it, table shouldn't exist if we met create table sql here

Comment (kennytm) [reply pkg/restore/db.go:58]: Ok.

File: tests/br_incompatible_tidb_config/run.sh (hunk 2 above)
Comment (3pointer) [reply tests/br_incompatible_tidb_config/run.sh:41]: okay, add test case below

File: tests/br_incompatible_tidb_config/run.sh [hunk 3]
```diff
@@ -29,17 +29,33 @@ run_sql "create schema $DB;"
...
+
+# drop pk
+run_sql "alter table $DB.$INCREMENTAL_TABLE drop primary key"
+run_sql "drop table $DB.$INCREMENTAL_TABLE"
+run_sql "create table $DB.$INCREMENTAL_TABLE like $DB.$TABLE"
+run_sql "insert into $DB.$INCREMENTAL_TABLE values (42, 42);"
+
+# incremental backup
+run_br --pd $PD_ADDR backup db --db "$DB" -s "local://$TEST_DIR/$DB$INCREMENTAL_TABLE"
```
Comment (kennytm) [inline tests/br_incompatible_tidb_config/run.sh:50]: this doesn't look incremental.

**Comment by 3pointer** [comment]
run-integration-test

**Pull request merged**